### PR TITLE
camera bugs now stick to things they're thrown at

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
camera bugs now stick to things they're thrown at, this applies to everything with the sticky component but just camera bugs get that as of now.

## Why It's Good For The Game
Adds a silly way to stick cameras to things. 

## Testing
Threw a camera bug at a skrell with throw mode on, worked as expected.

## Changelog
:cl:
tweak: camera bugs now stick to things they're thrown at
/:cl:

### CODE REVIEW

File: code/datums/components/sticky.dm [hunk 1]
```diff
@@ -69,6 +71,26 @@
 		START_PROCESSING(SSobj, src)
 	return COMPONENT_CANCEL_ATTACK_CHAIN
 
+/datum/component/sticky/proc/stick_to_it_throwing(obj/item/I, atom/hit_target, throwingdatum, params)
```
Comment (Gaxeer) [inline code/datums/components/sticky.dm:74]: what is `I` and `it` in `stick_to_it_throwing`

File: code/datums/components/sticky.dm [hunk 2]
```diff
@@ -69,6 +71,26 @@
...
+
+	attached_to.add_overlay(overlay, priority = TRUE)
+	I.invisibility = INVISIBILITY_ABSTRACT
+
+	RegisterSignal(attached_to, COMSIG_HUMAN_MELEE_UNARMED_ATTACKBY, PROC_REF(pick_up))
+	RegisterSignal(attached_to, COMSIG_PARENT_EXAMINE, PROC_REF(add_sticky_text))
+	RegisterSignal(attached_to, COMSIG_PARENT_QDELETING, PROC_REF(on_attached_destroy))
+	if(ismovable(attached_to))
+		RegisterSignal(attached_to, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
+		START_PROCESSING(SSobj, src)
```
Comment (Gaxeer) [inline code/datums/components/sticky.dm:92]: Code duplication. Should extract common code to proc

**Review by lewcc — CHANGES_REQUESTED** [review]
Please address Gaxeer's outstanding reviews

File: code/datums/components/sticky.dm (hunk 2 above)
Comment (GDNgit) [reply code/datums/components/sticky.dm]: Reviews applied

**Review by lewcc — APPROVED** [review]
LGTM

**Pull request merged**